nilearn/plotting/html_stat_map.py

```diff
@@ -1,157 +1,496 @@
+    data = encodebytes(handle_io.read()).decode('utf-8')
+    handle_io.close()
+    return data
+
+
+def _save_cm(output_cmap, cmap, format='png', n_colors=256):
+    """ Save the colormap of an image as an image file.
+    """
+
+    # save the colormap
+    data = np.arange(0., n_colors) / (n_colors - 1.)
+    data = data.reshape([1, n_colors])
+    imsave(output_cmap, data, cmap=cmap, format=format)
+
+
+def _sanitize_cm(cmap, n_colors=256):
+    """Make sure that the colormap has no duplicated colors.
+    """
+    # Extract a list of colors
+    cmap = plt.cm.get_cmap(cmap)
+    cmaplist = [cmap(i) for i in range(cmap.N)]
+
+    # Filter out any duplicated colors
+    mask = [cmaplist.count(cmap(i)) == 1 for i in range(cmap.N)]
+    if mask.count(True) < 2:
+        raise ValueError('The selected colormap has too many '
+                         'duplicated colors')
+    cmaplist = np.array(cmaplist)[np.array(mask)]
+
+    # Return a matplotlib colormap with the original number of colors
+    return LinearSegmentedColormap.from_list('Custom cmap', cmaplist, n_colors)
 
 
 class StatMapView(HTMLDocument):
     pass
 
 
-def _to_papaya(cmap, norm, abs_min, vmax, symmetric_cmap):
-    """Create colormaps ('lookup tables') given to papaya viewer"""
-    n_colors = 100
-    if not symmetric_cmap:
-        x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-        rgb = cmap(x)[:, :3]
-        return {'positive_cmap': _rgb_to_papaya(x, rgb),
-                'negative_cmap': None}
-    papaya = {}
-    x = np.linspace(norm(-vmax), norm(-abs_min), n_colors)
-    y = np.linspace(0, 1, n_colors)
-    rgb = cmap(x)[::-1, :3]
-    papaya['negative_cmap'] = _rgb_to_papaya(y, rgb)
-    x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-    rgb = cmap(x)[:, :3]
-    papaya['positive_cmap'] = _rgb_to_papaya(y, rgb)
-    return papaya
+def _mask_stat_map(stat_map_img, threshold=None):
+    """ Load a stat map and apply a threshold.
+    """
+    # Load stat map
+    stat_map_img = check_niimg_3d(stat_map_img, dtype='auto')
+
+    # Get data
+    data = _safe_get_data(stat_map_img, ensure_finite=True)
+
+    # threshold the stat_map
+    if threshold is not None:
+        data, threshold = _threshold_data(data, threshold)
+        mask_img = new_img_like(stat_map_img, data.mask, stat_map_img.affine)
+    else:
+        mask_img = new_img_like(stat_map_img, np.zeros(data.shape),
+                                stat_map_img.affine)
+    return mask_img, stat_map_img, data, threshold
+
+
+def _load_bg_img(stat_map_img, bg_img='MNI152', black_bg='auto', dim='auto'):
+    """ Resample bg_img in an isotropic resolution,
+        with a positive diagonal affine matrix.
+    """
+    # If no background is used, switch to the white background color style
+    if (bg_img is None or bg_img is False) and black_bg == 'auto':
+        black_bg = False
+
+    if bg_img is not None and bg_img is not False:
+        if isinstance(bg_img, str) and bg_img == "MNI152":
```

Comment:
again, `bg_img` can be an array, in which case `bg_img == "MNI152"` would not do what we want. hence the `isinstance(bg_img, str)`

Reply:
But the comparsion is using an `and`. If its is not 'MNI152' then that will be False.

Reply:
the left side of the `and` is evaluated first, if bg_img is not a string the rest is not evaluated

`bg_img is not False` will do what we want. bg_img == "MNI152" will return false only because scalar comparison will fail because of different types, otherwise it would return an array of False. in the future it will fail with an exception. bottomline: this is fine as it is